src/main/java/com/squareup/javapoet/ParameterizedTypeName.java

```diff
@@ -37,37 +42,101 @@
 
   ParameterizedTypeName(ClassName rawType, List<TypeName> typeArguments,
       List<AnnotationSpec> annotations) {
+    this(
+        rawType,
+        rawType.enclosingClassName() != null ? rawType.enclosingClassName() : rawType,
```

Comment:
If you do that, then we don't need a member field for nestedClassName, we can just do what you do on 48 when we need that